subprojects/platform-jvm/src/main/kotlin/org/gradle/kotlin/dsl/ToolchainManagementExtensions.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl
+
+import org.gradle.api.Incubating
+import org.gradle.api.toolchain.management.ToolchainManagement
+import org.gradle.jvm.toolchain.JvmToolchainManagement
+
+
+/**
+ * Provides statically defined accessors for configuring the "jvm" block in "toolchainManagement".
+ * This block is added to <code>ToolchainManagement</code> dynamically, by the "jvm-toolchain-management"
+ * plugin and normally the dynamic extension should generate accessors, but this doesn't work
+ * at the level of Settings, hence the need for the static accessors.
```

Comment:
:x: I would change this doc to provide useful information to the user only. The explanation about how this is done to circumvent some internal limitation is not useful, me thinks. A note that one needs to have the `jvm-toolchains` plugin applied in order for these to work would be better. Same with the other extension function below.